backend/corpora/common/entities/collection.py

```diff
@@ -280,3 +281,18 @@ def delete(self):
     def check_has_dataset(self, dataset: Dataset) -> bool:
         """Check that a dataset is part of the collection"""
         return all([self.id == dataset.collection_id, self.visibility == dataset.collection_visibility])
+
+    @staticmethod
+    def _normalize_doi(doi_url) -> str:
+        try:
+            return urlparse(doi_url.strip()).path.strip("/")
+        except Exception:
+            return None
+
+    def get_normalized_doi(self) -> str:
+        """Returns a normalized DOI if it exists, None otherwise"""
+        doi = [link for link in self.links if link.link_type == CollectionLinkType.DOI]
```

Comment:
Is it every possible for this to return more than one link?

Reply:
Publications should have a unique DOI so in theory, no. I don't think we're enforcing this though, so maybe we should do that when we add links.

Reply:
Can you create a ticket to ensue it is enforced? And maybe to double check the db to make sure nothing slipped through?